shared/js/background/pixel.es6.js

```diff
@@ -0,0 +1,102 @@
+/**
+ *
+ * This is part of our tool for anonymous engagement metrics
+ * Learn more at https://duck.co/help/privacy/atb
+ *
+ */
+
+const load = require('./load.es6')
+const browserWrapper = require('./$BROWSER-wrapper.es6')
+const settings = require('./settings.es6')
+const parseUserAgentString = require('../shared-utils/parse-user-agent-string.es6')
+
+/**
+ *
+ * Fire a pixel
+ *
+ * @param {string} pixelName
+ * @param {...*} args - any number of extra data
+ *
+ */
+function fire () {
+    if (!arguments.length) return
```

Comment:
We're making the assumption that `settings` is ready by the time this function is called. It's extremely unlikely that we'd call it _immediately_ on extension start, though, so it should be fine.

I don't think there's anything actionable here, just wanted to point it out.

Reply:
I wonder if you could trigger that by clicking on the extension menu bar item as soon as you can on first install, or when restoring a browsing session

Reply:
- Clicking asap on first install: tried, no problems
- Restoring a browsing session: didn't try, good one thanks

Reply:
@russellholt `settings` only needs to retrieve some stuff from extension local storage before it's ready, which is likely going to be somewhere in the millisecond range - much faster than a person can reasonably click the popup button.

The only way we can run into this problem is programmatically calling `fire()` somewhere from the background code immediately when the browser starts up, or when the extension is installed. I don't think we're likely to be doing that soon, though.